BusinessLogic/Entities/Auth/OAuthClientPassword.cs

```diff
@@ -117,15 +117,30 @@ public async Task<bool> SendEvent(Event e, Subscription subscription)
         /// <param name="response"></param>
         private void ResetAccessTokenIfErrorInResponseMessage(HttpResponseMessage response)
         {
-            List<Dictionary<string, string>> resultValues = JsonConvert.DeserializeObject<List<Dictionary<string, string>>>(response.Content.ReadAsStringAsync().Result);
 
-            if (resultValues.Count > 0)
+            // if error code string is there, then parse
+            if (response.Content.ReadAsStringAsync().Result.IndexOf("errorCode") != -1)
             {
-                Dictionary<string, string> firstItem = resultValues[0];
-                if (firstItem.ContainsKey("errorCode") && firstItem["errorCode"] == "INVALID_SESSION_ID")
+                // try to parse the json to find the error code
+                try
                 {
-                    Log.Debug($"OAuthClientPassword.ResetAccessTokenIfErrorInResponseMessage: AccessToken set to `null` ");
-                    AccessToken = null;
+                    List<Dictionary<string, string>> resultValues = JsonConvert.DeserializeObject<List<Dictionary<string, string>>>(response.Content.ReadAsStringAsync().Result);
+
+                    if (resultValues.Count > 0)
+                    {
+                        Dictionary<string, string> firstItem = resultValues[0];
+                        if (firstItem.ContainsKey("errorCode") && firstItem["errorCode"] == "INVALID_SESSION_ID")
+                        {
+                            Log.Debug($"OAuthClientPassword.ResetAccessTokenIfErrorInResponseMessage: AccessToken set to `null` ");
+                            AccessToken = null;
+                        }
+                    }
+                }
+                catch (JsonReaderException)
+                {
+                    // JsonConvert fails because the Dictionary<string, string> is not met,
+                    // ergo there is no error but the normal data
+                    // so nothing to do here, just continue
```

Comment:
So just to clarify, this fails when the body is an empty string or null?

Reply:
No, this fails if it has data, **but** the structure does not fit the Dictionary<string, string> structure, and this would be because the server responded correctly with a totally different structure, or the expected result